File: third-year/matlab-simulations/MultiplicativeCongruentialMethod.cpp
#include <iostream>
using namespace std;
int main()
{
    int a, m, r, n;
    cout << "Enter the value of a:\n";
    cin >> a;
    cout << "Enter the value of m:\n";
    cin >> m;
    cout << "Enter the value of seed or r:\n";
    cin >> r;
    cout << "Enter the total random numbers to be generated:\n";
    cin >> n;
    cout << "The " << n << " random numbers generated are:\n";
    for (int i = 0; i < n; i++)
    {
        r = (a * r) % m;
        cout << r << " ";
    }
    cout << "\n";
    return 0;
}

File: third-year/matlab-simulations/multiplicative_congruential_method.m
%% r(i+1) = (ar(i)) mod m
clc;
a = input('Please Enter the value of a : ');
m = input('Please Enter the value of m : ');
r = input('Please Enter the value of seed or r : ');

totalNumber = input('Number of Random Numbers want to generate : ');

for i = 1:totalNumber
    r = mod(a * r, m);
    fprintf('%4.0f ', r);
end

fprintf('\n');
